system/lib/libc/musl/src/thread/pthread_cancel.c

```diff
@@ -3,7 +3,11 @@
 #include "pthread_impl.h"
 #include "syscall.h"
 
+#ifdef __EMSCRIPTEN__
+hidden long __cancel();
+#else
 hidden long __cancel(), __syscall_cp_asm(), __syscall_cp_c();
+#endif
```

Comment:
I don't think this patch should be needed .. declaring extra stuff is petty harmless.

Reply:
Reverted that patch with commit https://github.com/emscripten-core/emscripten/pull/15603/commits/73ed3d4a60d7278e8d277d8e860a4d9293366802 in PR #15603.

Reply:
BTW, GitHub does not allow me to stack PRs when I use my own fork to base my branch on (instead of pushing the branches to the main repository). Hence, this PR is currently easiest to review per commit.